arkane/ess/gaussian.py

```diff
@@ -57,8 +57,47 @@ class GaussianLog(ESSAdapter):
     arrays. GaussianLog is an adapter for the abstract class ESSAdapter.
     """
 
-    def __init__(self, path):
+    def __init__(self, path, check_for_errors=True):
         self.path = path
+        if check_for_errors:
+            self.check_for_errors()
+
+    def check_for_errors(self):
+        """
+        Checks for common errors in a Gaussian log file.
+        If any are found, this method will raise an error and crash.
+        """
+        with open(os.path.join(self.path), 'r') as f:
+            lines = f.readlines()
+            error, keywords = None, None
+            for line in reversed(lines):
+                # check for common error messages
+                if 'termination' in line:
```

Comment:
I don't find an `else` here. Generally, without a `termination` shown up in the file. It is highly possible that the job is not finished. Besides, according to my memory, for a single job, `termination` will only show up once at the very end; but for a complex job (e.g., composite methods or opt freq jobs) it may appear at several places. So better only check the term `termination` at the last few lines (e.g. lines[-1] to lines[-100]). Also you can add a `break` once find the identifier to avoid going further.

Reply:
Thanks for the suggestions. I'll update this to only iterate through the last 100 lines in the log file and to break if the file terminated without errors. I pushed a fixup commit. Let me know if I should modify it further or if it's okay to squash